internal/mode/static/state/change_processor.go

```diff
@@ -222,7 +222,7 @@ func NewChangeProcessorImpl(cfg ChangeProcessorConfig) *ChangeProcessorImpl {
 			{
 				gvk:       cfg.MustExtractGVK(&ngfAPI.SnippetsFilter{}),
 				store:     newObjectStoreMapAdapter(clusterStore.SnippetsFilters),
-				predicate: nil, /*TODO(kate-osborn): will add predicate in next PR*/
+				predicate: nil, // we always want to write status to SnippetsFilters so we don't filter them out
```

Comment:
I debated this for a long time. Generally, we don't write status to resources that aren't referenced by a Gateway object. I think SnippetsFilters should be an exception because we expect the owner of the SnippetsFilter to be a cluster operator and the owner of the Route that references the SnippetsFilter to be an app dev. The cluster operator will want to apply the SnippetsFilter first and make sure it's accepted before the app dev references it. Therefore, I think we want to always write statuses to SnippetsFilters.

Reply:
I think that's fair, since there are implications of this resource unlike other policies.